pkg/cmd/accept/exec.go

```diff
@@ -61,95 +64,133 @@ func (o *Options) run() error {
+	csrApproved, err := o.approveCSR(kubeClient, clusterName)
+	if err != nil {
+		return false, err
+	}
+	mcUpdated, err := o.updateManagedCluster(clusterClient, clusterName)
+	if err != nil {
+		return false, err
+	}
+	if csrApproved && mcUpdated {
+		return true, nil
+	}
+	return false, nil
+
+}
+
+func (o *Options) approveCSR(kubeClient *kubernetes.Clientset, clusterName string) (bool, error) {
+	csrs, err := kubeClient.CertificatesV1().CertificateSigningRequests().List(context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%v = %v", clusterLabel, clusterName),
+		})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	var csr *certificatesv1.CertificateSigningRequest
+	for _, item := range csrs.Items {
+		//Does not have the correct name prefix
+		if !strings.HasPrefix(item.Spec.Username, userNameSignaturePrefix) {
+			continue
+		}
+		//Check groups
+		var group string
+		for _, g := range item.Spec.Groups {
+			if g == groupName {
+				group = g
+				break
 			}
-			//Check if already approved or denied
-			done := false
-			for _, c := range item.Status.Conditions {
-				if c.Type == certificatesv1.CertificateApproved || c.Type == certificatesv1.CertificateDenied {
-					done = true
-					break
-				}
+		}
+		//Does not contain the correct group
+		if len(group) == 0 {
+			continue
+		}
+		//Check if already approved or denied
+		for _, c := range item.Status.Conditions {
+			if c.Type == certificatesv1.CertificateApproved {
+				fmt.Printf("CSR %s already approved\n", item.Name)
+				return true, nil
 			}
-			if done {
-				continue
+			if c.Type == certificatesv1.CertificateDenied {
+				fmt.Printf("CSR %s already denied\n", item.Name)
+				return true, nil
 			}
-			csr = &item
-			break
 		}
+		csr = &item
+		break
+	}
 
-		if csr != nil {
-			if !o.ClusteradmFlags.DryRun {
-				if csr.Status.Conditions == nil {
-					csr.Status.Conditions = make([]certificatesv1.CertificateSigningRequestCondition, 0)
-				}
-
-				csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1.CertificateSigningRequestCondition{
-					Status:         corev1.ConditionTrue,
-					Type:           certificatesv1.CertificateApproved,
-					Reason:         fmt.Sprintf("%sApprove", helpers.GetExampleHeader()),
-					Message:        fmt.Sprintf("This CSR was approved by %s certificate approve.", helpers.GetExampleHeader()),
-					LastUpdateTime: metav1.Now(),
-				})
-
-				kubeClient, err := o.ClusteradmFlags.KubectlFactory.KubernetesClientSet()
-				if err != nil {
-					return err
-				}
-				signingRequest := kubeClient.CertificatesV1().CertificateSigningRequests()
-				if _, err := signingRequest.UpdateApproval(context.TODO(), csr.Name, csr, metav1.UpdateOptions{}); err != nil {
-					return err
-				}
+	if csr != nil {
+		if !o.ClusteradmFlags.DryRun {
```

Comment:
this seems weird....I think the whole func should not be called with dryrun?

Reply:
No in dry-run,  we will not approve but at a least we will check if csr is present.

Reply:
then it might be

```
if csr == nil {
    fmt.Printf("no CSR to approve for cluster %s\n", clusterName)
    return false, nil
}
if o.ClusteradmFlags.DryRun {
    return true, nil
}

// approve csr
```

Reply:
ok more simple ;-)